packages/stream_chat_flutter_core/lib/src/channels_bloc.dart

```diff
@@ -95,7 +95,12 @@ class ChannelsBlocState extends State<ChannelsBloc>
   }) async {
     final client = StreamChatCore.of(context).client;
 
-    if (_paginationEnded || _queryChannelsLoadingController.value == true) {
+    final clear = paginationParams == null ||
+        paginationParams.offset == null ||
+        paginationParams.offset == 0;
+
+    if ((!clear && _paginationEnded) ||
+        _queryChannelsLoadingController.value == true) {
```

Comment:
I don't think we really to do this for the fix

Reply:
https://github.com/GetStream/stream-chat-flutter/pull/414#discussion_r626516062